build.sh

```diff
@@ -6,10 +6,21 @@
 
 set -ex
 
-if [ "$1" == "--help" ] || [ "$1" == "-h" ]; then
-    echo "--llvm to rebuild llvm";
-    exit;
-fi
+WITH_NIX=
+REBUILD_LLVM=
+case "$1" in
+    --nix)
+        WITH_NIX=1
+        shift
+        ;;
+    --llvm)
+        REBUILD_LLVM=1
+        shift
+        ;;
+    --help)
+        echo "--llvm to rebuild llvm, --nix to use nix";
+        exit;
+esac
```

Comment:
This means `--nix` and `--llvm` can't be used together. I don't think this is much of a hassle, so if you are OK with that, we can merge as it is.

Reply:
Oops sorry, you're right, wrapped this in a while loop!